tests/admin_views/tests.py

```diff
@@ -2272,6 +2273,103 @@ def test_post_save_message_no_forbidden_links_visible(self):
         )
 
 
+@override_settings(
+    ROOT_URLCONF='admin_views.urls',
+    # Test with the admin's documented list of required context processors.
+    TEMPLATES=[{
+        'BACKEND': 'django.template.backends.django.DjangoTemplates',
+        'APP_DIRS': True,
+        'OPTIONS': {
+            'context_processors': [
+                'django.contrib.auth.context_processors.auth',
+                'django.contrib.messages.context_processors.messages',
+            ],
+        },
+    }],
+)
+class AdminViewProxyModelPermissionsTest(TestCase):
```

Comment:
It's not obvious to me why these tests are needed. Is there a particular concern about a bug in the admin such that this wouldn't work?

Reply:
@carltongibson TL;DR; Yes this is expected.

We should see 4 errors indeed but I think it's because it fails on `setUpTestData`, therefor none of the tests of the test case are run:
```
$ ./runtests.py admin_views.tests.AdminViewProxyModelPermissionsTest

======================================================================
ERROR: setUpClass (admin_views.tests.AdminViewProxyModelPermissionsTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/arthur/code/django/django/test/testcases.py", line 1036, in setUpClass
    cls.setUpTestData()
  File "/Users/arthur/code/django/tests/admin_views/tests.py", line 2417, in setUpTestData
    cls.viewuser.user_permissions.add(get_perm(UserProxy, get_permission_codename('view', opts)))
  File "/Users/arthur/code/django/tests/admin_views/tests.py", line 1369, in get_perm
    return Permission.objects.get(content_type=ct, codename=codename)
  File "/Users/arthur/code/django/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/Users/arthur/code/django/django/db/models/query.py", line 408, in get
    self.model._meta.object_name
django.contrib.auth.models.Permission.DoesNotExist: Permission matching query does not exist.
```

If you revert both changes (`get_perm` and `contrib/auth/management/__init__.py`), then `get_perm` behaves the same as in `contrib/auth/management/__init__.py` so the `setUpTestData` doesn't raise an error and we get 4 failures instead of 1 error.